File: src/main/java/ankol/mod/merger/core/ParsedResult.java
package ankol.mod.merger.core;

import org.antlr.v4.runtime.TokenStream;

/**
 * 语法树解析结果
 *
 * @param astNode     解析后的AST节点
 * @param tokenStream 原始Token流
 * @author Ankol
 */
public record ParsedResult<T extends BaseTreeNode>(T astNode, TokenStream tokenStream) {
}

File: src/main/java/ankol/mod/merger/core/ParsedResult.kt
package ankol.mod.merger.core

import org.antlr.v4.runtime.TokenStream

/**
 * 语法树解析结果
 * 
 * @param astNode     解析后的AST节点
 * @param tokenStream 原始Token流
 * @author Ankol
 */
data class ParsedResult<N : BaseTreeNode>(val astNode: N, val tokenStream: TokenStream)
